setuptools_rust/build.py

```diff
@@ -162,12 +167,19 @@ def build_extension(self, ext):
                 args.extend(
                     ["-C", "link-arg=-undefined", "-C", "link-arg=dynamic_lookup"]
                 )
+            # Tell musl targets not to statically link libc. See
+            # https://github.com/rust-lang/rust/issues/59302 for details.
+            if b'target_env="musl"' in rust_target_info:
+                rustflags += " -C target-feature=-crt-static"
```

Comment:
Should this be `args.extend(["-C", "target-feature=-crt-static"])`?

Reply:
I tried that. It didn't work. I don't know why :-)

Reply:
Strange. What output did you get from the error?

From `cargo rustc` docs:

```
This command requires that only one target is being compiled. If more than one
target is available for the current package the filters of --lib, --bin, etc,
must be used to select which target is compiled. To pass flags to all compiler
processes spawned by Cargo, use the $RUSTFLAGS environment variable or the
`build.rustflags` configuration option.
```

So I _think_ that adding the args shoud be equivalent to setting `RUSTFLAGS`, but with the upside that we can `print` the full invocation correctly...

Reply:
I get an identical output to if I didn't pass the value at all in terms of error message. The CLI invocation that's printed does show that the args are passed.

Reply:
Example output:

```
cargo rustc --lib --manifest-path src/rust/Cargo.toml --features pyo3/extension-module --release --verbose -- -C target-feature=-crt-static --crate-type cdylib
error: cannot produce cdylib for `cryptography-rust v0.1.0 (/cryptography/src/rust)` as the target `x86_64-unknown-linux-musl` does not support these crate types
error: cargo failed with code: 101
```

Reply:
Oh, found it:
> The specified ARGS will all be passed to the final compiler invocation, not any of the dependencies
this explains why both this and the target-cpu need to go in rustflags